app/dependencies.php

```diff
@@ -234,6 +237,24 @@ public function __invoke(array $record)
             return $logger;
         },
 
+        $donationCreateRateLimiterKey => function (ContainerInterface $c): RateLimiterFactory {
+            return new RateLimiterFactory(
+                config: [
+                    'id' => 'create-donation',
+                    'policy' => 'token_bucket',
```

Comment:
I've looked at the docs but am still confused as to what the default 'limiter' that's in effect here will be. Is it based on the direct client IP? If so that will be a problem because we're behind a load balancer.

It might also be worth thinking about the relative advantages of using a new limited as we have solved this problem once already via middleware like at https://github.com/thebiggive/matchbot/blob/824c343503f9862c1964a356b56a2348fa5cd95d/app/routes.php#L39-L40

Reply:
No it won't be based on client IP, it will be based on the key passed to `\Symfony\Component\RateLimiter\RateLimiterFactory::create`  in `createDonation` is the psp customer ID.

Reply:
Thanks and sorry, missed the key line! I've added a table in 8bdbdec672b5cc3899c04ddc2f4cd01dad42c3ff with a high level overview to hopefully make it easier to understand the approaches used through the app.

Reply:
Welcome, no need to apologies.